2009-10-14  Michael Tautschnig  <mt@debian.org>

	* MT: sync changelog, 3.2.23 already contains entry about bootable!
	* setup-storage/Parser.pm: Removed "mark / as bootable" code
	* man/setup-storage.8: Document possible problems with previous software RAID
		partitions and missing bootable flags.
Index: trunk/lib/setup-storage/Parser.pm
===================================================================
--- trunk.orig/lib/setup-storage/Parser.pm
+++ trunk/lib/setup-storage/Parser.pm	
@@ -529,12 +529,6 @@
           } else {
             $FAI::partition_pointer->{encrypt} = 0;
           }
-          if ($FAI::device =~ /^PHY_(.+)$/ &&
-            -1 == $FAI::configs{$FAI::device}{bootable} &&
-            $FAI::partition_pointer->{mountpoint} eq "/") {
-            $FAI::configs{$FAI::device}{bootable} =
-              $FAI::partition_pointer->{number};
-          }
         }
 
     name: m{^([^/,\s\-]+)}
Index: trunk/man/setup-storage.8
===================================================================
--- trunk.orig/man/setup-storage.8
+++ trunk/man/setup-storage.8	
@@ -416,6 +416,22 @@
 .sp
 .fi
 .PP
+.SH CAVEATS
+.IP \(bu
+Partition UUID cannot be obtained: In case a partition was previously used as
+part of a software RAID volume and now is intended as swap space, udev fails
+when asked for a UUID. This happens because mkswap does not overwrite the
+previous RAID superblock. You can remove it using mdadm --zero-superblock
+<device>.
+.IP \(bu
+Machine does not boot because not partition is marked as bootable: If the
+bootable option is not specified, not partition will be marked as such. Modern
+BIOSes don't seem to require such markers anymore, but for some systems it may
+still be necessary. Previous versions of setup-storage by default marked the
+partition mounting / as bootable, but this is not a sane default for all cases.
+If you want to be sure not boot failures happen because of a missing bootable
+marker, explicitly set the bootable option. Of course, there are lots of other
+reasons why a system may fail to boot.
 .SH SEE ALSO
 This program is part of FAI (Fully Automatic Installation).
 The FAI homepage is http://www.informatik.uni-koeln.de/fai.
